siliconcompiler/tools/yosys/syn_asic.py

```diff
@@ -203,7 +203,7 @@ def prepare_synthesis_libraries(chip):
         f.write(markDontUse.processLibertyFile(
             dff_liberty_file,
             dff_dont_use,
-            chip.get('option', 'quiet', step=step, index=index),
+            logger=None if chip.get('option', 'quiet', step=step, index=index) else chip.logger
```

Comment:
You could move this outside the `with open(` and then later reuse it.
Then you don't have this unwieldy ternary expression inside here and the call down there.

Reply:
I'm not sure I follow your suggestion